go/vt/vttablet/tabletserver/tabletenv/config.go

```diff
@@ -183,7 +183,7 @@ func registerTabletEnvFlags(fs *pflag.FlagSet) {
 
 	fs.BoolVar(&enableHeartbeat, "heartbeat_enable", false, "If true, vttablet records (if master) or checks (if replica) the current time of a replication heartbeat in the sidecar database's heartbeat table. The result is used to inform the serving state of the vttablet via healthchecks.")
 	fs.DurationVar(&heartbeatInterval, "heartbeat_interval", 1*time.Second, "How frequently to read and write replication heartbeat.")
-	fs.DurationVar(&heartbeatOnDemandDuration, "heartbeat_on_demand_duration", 0, "If non-zero, heartbeats are only written upon consumer request, and only run for up to given duration following the request. Frequent requests can keep the heartbeat running consistently; when requests are infrequent heartbeat may completely stop between requests")
+	fs.DurationVar(&heartbeatOnDemandDuration, "heartbeat_on_demand_duration", 0, "If non-zero, heartbeats are only written upon consumer request, and only run for up to given duration following the request. Frequent requests can keep the heartbeat running consistently. Automatically set to 5s when --heartbeat_enable is unset.")
```

Comment:
That's a great idea, irrespective of this PR! I only wonder if it's going to break someone's existing setup?

Reply:
Yeah, it might... 😢 We could potentially add it to the 19.0.0 changelog as a breaking change though.

Maybe we should take a documentation driven approach here. How would we document the new/proposed configuration here? https://vitess.io/docs/19.0/reference/features/tablet-throttler/#configuration

1. IMO we should remove all of the old tablet level config parts/noise there in the v19 docs
2. That seems a little misleading now as it says "Enabling the lag throttler also automatically enables heartbeat injection." and it does not note that you must set `--heartbeat_enable=false` (which I think was true before given the noted issues reported with the examples) and `--heartbeat_on_demand_duration` to a non-zero value — although it does describe what it is and recommend values for it